### ISSUE TITLE
ERROR: modpost on ArmV7 SBCs

### ISSUE THREAD

**Issue by crackerjacques** [bondagit/ravenna-alsa-lkm#19]
Hello.

It is stated that it works fine with Arm v7.
However, I have tried with the following SBCs
```
RPI Zero 
RPI Zero 2W
Luckfox Pico Max
```

```
ERROR: modpost: "__aeabi_uldivmod" [/home/jackheinlein/aes67-linux-daemon/3rdparty/ravenna-alsa-lkm/driver/MergingRavennaALSA.ko] undefined!
ERROR: modpost: "__aeabi_ldivmod" [/home/jackheinlein/aes67-linux-daemon/3rdparty/ravenna-alsa-lkm/driver/MergingRavennaALSA.ko] undefined!
make[2]: *** [scripts/Makefile.modpost:126: /home/jackheinlein/aes67-linux-daemon/3rdparty/ravenna-alsa-lkm/driver/Module.symvers] エラー 1
make[1]: *** [Makefile:1964: modpost] エラー 2
make[1]: ディレクトリ '/usr/src/linux-headers-6.1.21-v7+' から出ます
make: *** [Makefile:15: modules] エラー 2
```

This is the error.
I have tried to edit the Makefile in various ways, but it is not working in the end.

It is Japanese in places, but these mean.
"エラー" means "error".
"ディレクトリ" means "directory".
"から出ます" means "out form ~~~ "

I don't know much about kernel modules, so if you have any solutions, I would be grateful.
kindly.

**Comment by djsakisd** [comment]
@crackerjacques try [tag 1.7 ](https://github.com/bondagit/ravenna-alsa-lkm/releases/tag/v1.7 )
@bondagit  I also have the same issue with raspberry 32bit - tried in 5.10 & 6.x kernels  on raspbian buster & bookworm accordingly with same issue.

**Comment by crackerjacques** [comment]
Thanks a lot.
I will try it.

**Comment by bondagit** [comment]
I have pushed a fix in [branch issue19](https://github.com/bondagit/ravenna-alsa-lkm/tree/issue19). Can you check if this works for you ?

**Comment by crackerjacques** [comment]
> I have pushed a fix in [branch issue19](https://github.com/bondagit/ravenna-alsa-lkm/tree/issue19). Can you check if this works for you ?

okay just a moment.
i will try it too.

and tag 1.7 is worked.

**Comment by djsakisd** [comment]
@bondagit Sorry for the late reply i didn't have a device available,  i just tested it and It compiles and works fine now.
Raspbian Buster on  5.10.103-v7l+

**Comment by crackerjacques** [comment]
> I have pushed a fix in [branch issue19](https://github.com/bondagit/ravenna-alsa-lkm/tree/issue19). Can you check if this works for you ?

That worked for me too. Thank you very much.

on  Ubuntu22.4.3 / Luckfox Pico Pro Max 
![2024-06-08_21-46-29](https://github.com/bondagit/ravenna-alsa-lkm/assets/129548912/9ed39cf8-dbb9-4ff4-b1b7-a64594ea5d0c)


However, with Luckfox's limited processor and RAM, all I could do was load .ko and some simple test.
![2024-06-08_21-46-29](https://github.com/bondagit/ravenna-alsa-lkm/assets/129548912/f95aeebe-64ac-40f3-bc56-642d96ef34d0)

**Comment by bondagit** [comment]
Merged into driver v1.10
